production.py

```diff
@@ -337,7 +337,7 @@ def calc_group(self, type_, kind):
 
         for move in self.costs:
             if (not move.stock_move or move.type_ != type_ or move.kind != kind
-                    or move.stock_move.state == 'cancelled'):
+                    or move.stock_move.state == 'cancelled' and not move.quantity):
```

Comment:
In case is 0 (zero), return calc group (create a new move) or continue next cost?

Reply:
I don't know the code well enough but not that at line 370 the divisor is `move.quantity + move_cost.quantity` not just `move.quantity`.

Reply:
The traceback is:

```
  File "/tryton/tryton/trytond/trytond/modules/production_cost_analysis/production.py", line 370, in calc_group
    (float(move.cost_price) * move.quantity
ZeroDivisionError: float division by zero
```

Line 365 and 376, set quantity from move:

```
move_cost.quantity = move.quantity
...
res[move.product] = move_cost
```

and finally, move_cost is get from the "res" dict:

```
move_cost = res.get(move.product)
``

Reply:
The change is not correct.

In this case, the sum they have of the same product in different productions is being calculated, which will finally be compared with the same productions in other states.

Therefore we are interested in that if anything there is only a movement to zero it is reflected, and as it is we avoid it.

we need to make the change on the line checking that yes
zero,
in line 369:
if move.quantity + move_cost.quantity is equal, we can leave move.cost_price

Reply:
See changeset dcd7c13bb6b6fa886b3c8903d541afc440051882